app/javascript/mastodon/service_worker/web_push_notifications.js

```diff
@@ -57,12 +57,16 @@ const openUrl = url =>
         .filter(client => /\/web\//.test(client.url))
         .sort(client => client !== 'visible');
 
-      const visibleClient = clientList.find(client => client.visibilityState === 'visible');
-      const focusedClient = clientList.find(client => client.focused);
+      if (webClients.length !== 0) {
+        return webClients[0].focus().then(client => client.postMessage({ type: 'navigate', url }));
+      } else {
+        const visibleClient = clientList.find(client => client.visibilityState === 'visible');
+        const focusedClient = clientList.find(client => client.focused);
```

Comment:
not sure if this is the desired behavior. why fall back to navigating something that's not the webClient? if I have a profile page open, won't this hijack it? is that desired?